File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_a2/organic3_bubble_sort.java
/*
 * Bubble Sort - organic3
 * Approach: While loop tracking swapped flag until no swaps occur.
 */

import java.util.Arrays;

class BubbleSort {
    public static int[] bubbleSort(int[] arr) {
        int n = arr.length;
        boolean swapped = true;
        while (swapped) {
            swapped = false;
            for (int i = 0; i < n - 1; i++) {
                if (arr[i] > arr[i + 1]) {
                    int temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                    swapped = true;
                }
            }
            n--;
        }
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90, 5, 77, 1};
        System.out.println("Sorted array: " + Arrays.toString(bubbleSort(arr)));
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_a2/organic3_bubble_sort.py
"""
Bubble Sort - organic3
Approach: Uses a `swapped` flag to exit early once a full pass makes
no swaps, wrapped in a while loop instead of a fixed range of passes.
"""


def bubble_sort(arr):
    n = len(arr)
    swapped = True
    while swapped:
        swapped = False
        for i in range(n - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                swapped = True
        n -= 1
    return arr


if __name__ == "__main__":
    ARR = [64, 34, 25, 12, 22, 11, 90, 5, 77, 1]
    print("Sorted array:", bubble_sort(ARR))
